### PR TITLE
Make Profile command compatible with IDs 

### PR DESCRIPTION
**Pull request by dkim19375**
Allows you to use the profile command of someone else without pinging them :)

Btw this is like my first time using kotlin coroutines so idk if this works lol

### CODE REVIEW

**Comment by dkim19375** [comment]
smh codacy

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt
Comment (knightzmc) [inline src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:19]: Probably want to use run {} here, rather than let {}

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt
Comment (knightzmc) [inline src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:18]: Why use a callback rather than just returning the Member?

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt
Comment (knightzmc) [inline src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:39]: Unnecessary return

**Review by knightzmc — CHANGES_REQUESTED** [review]
Overall seems good, although there seem to be a few questionable decisions, and so this could be simplified a lot

Have a look at my review comments - as an extension, I don't think you should need to modify RestAction.await at all

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt
Comment (dkim19375) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt]: fixed

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt
Comment (dkim19375) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt]: i just did `?: return null`

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt
Comment (knightzmc) [inline src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:23]: why is this needed?

**Review by knightzmc — CHANGES_REQUESTED** [review]

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt
Comment (dkim19375) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:23]: wdym? in case the user isn't cached

Comment (knightzmc) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:23]: i mean the new await function - can't you just do `return guild.retrieveMemberById(id.toString(), false).await()`?

Comment (dkim19375) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:23]: in case it errors due to no user found... 
https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/entities/Guild.html#retrieveMemberById(long)
If the user is invalid, it errors

Comment (knightzmc) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:23]: ah makes sense, maybe a try / catch would be nicer but this seems alright

**Comment by knightzmc** [comment]
codacy bad

**Comment by dkim19375** [comment]
btw dont merge yet

**Comment by dkim19375** [comment]
also why's the profile command in the xp package, also the commands package confused me because no command handling was there :p

**Comment by knightzmc** [comment]
@dkim19375 it would seem that this doesnt compile

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt
Comment (dkim19375) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:23]: > @dkim19375 it would seem that this doesnt compile

that's why i said don't merge :))

Comment (dkim19375) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/Messages.kt:23]: why did it reply here

**Comment by knightzmc** [comment]
> also why's the profile command in the xp package, also the commands package confused me because no command handling was there :p

yeah the packaging is a bit inconsistent at the moment - from now on we should try and package by feature (i.e XPCommand in the xp package not commands package)

**Comment by dkim19375** [comment]
> @dkim19375 it would seem that this doesnt compile

That's why I said don't merge yet :))

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/RestActions.kt
Comment (knightzmc) [inline src/main/kotlin/me/bristermitten/devdenbot/extensions/RestActions.kt:14]: if we're being really petty the types don't need to be specified here, might be worth changing to make it a bit cleaner

**Review by knightzmc — APPROVED** [review]

**Comment by knightzmc** [comment]
Ready to merge now unless you want to implement that last comment I made

File: src/main/kotlin/me/bristermitten/devdenbot/extensions/RestActions.kt
Comment (dkim19375) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/RestActions.kt:14]: Seems to error - `Cannot infer a type for this parameter. Please specify it explicitly.`
Although it's a simple fix by adding <T> i think we should just keep the return type.. idk, for clarity, although I will commit a code cleanup change :P

Comment (knightzmc) [reply src/main/kotlin/me/bristermitten/devdenbot/extensions/RestActions.kt]: i dont mean the return type, i mean the type of `continuation` and `throwable`

**Review by knightzmc — COMMENTED** [review]
Nice - I wonder if you could do `failure: (Continuation<T>, Throwable) -> Unit = Continuation<T>::resumeWithException`?

**Comment by dkim19375** [comment]
uh

**Comment by dkim19375** [comment]
Errors - `<html>Type inference failed: inline fun &lt;T&gt; Continuation&lt;T#1 (type parameter of kotlin.coroutines.resumeWithException)&gt;.resumeWithException(exception: Throwable): Unit<br/>cannot be applied to<br/>receiver: Continuation&lt;T#2 (type parameter of me.bristermitten.devdenbot.extensions.await)&gt;  arguments: ()`
uhhhhhhhhhhhhhhhhhhhh dont ask why it has all those tags

**Comment by knightzmc** [comment]
darn kotlin bad

**Comment by knightzmc** [comment]
okay this is fine then :))

**Comment by dkim19375** [comment]
:o

**Comment by dkim19375** [comment]
woohoo im a contributer 🥳

**Comment by dkim19375** [comment]
now when will you restart the bot 👀

**Comment by knightzmc** [comment]
it's building automatically, give it a minute

**Pull request merged**
